application/common/dos.py

```diff
@@ -115,15 +115,27 @@ def get_matching_dos_services(odscode: str, org_type_id: str) -> List[DoSService
         list[DoSService]: List of DoSService objects with matching first 5
         digits of odscode, taken from DoS database
     """
-    odscode = get_new_odscode_for_dos(odscode, org_type_id)
-    logger.info(f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode}'")
-    sql_query = (
-        "SELECT s.id, uid, s.name, odscode, address, town, postcode, web, email, fax, nonpublicphone, typeid,"
-        " parentid, subregionid, statusid, createdtime, modifiedtime, publicphone, publicname, st.name servicename"
-        " FROM services s LEFT JOIN servicetypes st ON s.typeid = st.id"
-        " WHERE odscode LIKE %(ODS)s"
-    )
-    named_args = {"ODS": f"{odscode}%"}
+    odscode_5_or_6 = get_new_odscode_for_dos(odscode, org_type_id)
+    if org_type_id == PHARMACY_ORG_TYPE_ID:
+        logger.info(f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode_5_or_6}'")
+        sql_query = (
+            "SELECT s.id, uid, s.name, odscode, address, town, postcode, web, email, fax, nonpublicphone, typeid,"
+            " parentid, subregionid, statusid, createdtime, modifiedtime, publicphone, publicname, st.name servicename"
+            " FROM services s LEFT JOIN servicetypes st ON s.typeid = st.id"
+            " WHERE odscode LIKE %(ODS)s"
+        )
+        named_args = {"ODS": f"{odscode_5_or_6}%"}
+    elif org_type_id == DENTIST_ORG_TYPE_ID:
+        logger.info(
+            f"Searching for '{org_type_id}' DoS services with ODSCode that matches '{odscode[0:7]} or {odscode_5_or_6}'"
+        )
+        sql_query = (
+            "SELECT s.id, uid, s.name, odscode, address, town, postcode, web, email, fax, nonpublicphone, typeid,"
+            " parentid, subregionid, statusid, createdtime, modifiedtime, publicphone, publicname, st.name servicename"
+            " FROM services s LEFT JOIN servicetypes st ON s.typeid = st.id"
+            " WHERE odscode LIKE %(ODS)s or odscode LIKE %(ODS6)s"
+        )
+        named_args = {"ODS": f"{odscode[0:7]}%", "ODS6": f"{odscode_5_or_6}%"}
```

Comment:
I've had a conversation with Nicole. So NHS UK pharmacies are always 7 characters long which must be matched to DoS 6 characters (minus a 0 from NHS UK pharmacy) and also 7 or longer ods codes. So for this scenario this SQL query doesn't make sense.
![image](https://user-images.githubusercontent.com/62281988/162224217-a2c4a7d8-8ef6-46c3-945a-5a200b4d0b07.png)

Reply:
Yes, i will update the code